lua/customroles/elementalist/cl_elementalist.lua

```diff
@@ -0,0 +1,175 @@
+    overlayPanel:SetSize(IceOverlay:GetWide(), IceOverlay:GetTall())
+    overlayPanel:SetPos(0, 0)
+    overlayPanel.Paint = function()
+        if alphaCounter <= 125 then alphaCounter = alphaCounter + 0.5 end
+        surface.SetDrawColor(255, 255, 255, alphaCounter)
+        surface.SetMaterial(IceOverlayMat)
+        surface.DrawTexturedRect(0, 0, overlayPanel:GetWide(), overlayPanel:GetTall())
+    end
+end
+
+function CloseIceOverlay()
+    if IceOverlay and ispanel(IceOverlay) then IceOverlay:Remove() end
+end
+
+function StartBlindOverlay()
+    DarkOverlay = {
+        ["$pp_colour_brightness"] = -1,
+        ["$pp_colour_colour"] = 0
+    }
+end
+
+function StartDarkOverlay(percent)
+    local actualPercent = percent * 0.01
+
+    shouldDrawDarkOverlay = true
+
+    DarkOverlay = {
+        ["$pp_colour_brightness"] = -0.25 * actualPercent,
+        ["$pp_colour_colour"] = 1 - (1 * actualPercent)
+    }
+end
+
+function EndDarkOverlay()
+    shouldDrawDarkOverlay = false
+
+    DarkOverlay = {
+        ["$pp_colour_brightness"] = 0,
+        ["$pp_colour_colour"] = 1
+    }
+end
+
+hook.Add("RenderScreenspaceEffects", "ElementalistScreenDimming", function()
+    if shouldDrawDarkOverlay then
+        DrawColorModify(DarkOverlay)
+    end
+end)
+
+net.Receive("BeginIceScreen", function(len)
+    local frozen = net.ReadBool()
+
+    CreateIceOverlay(frozen)
+end)
+
+net.Receive("EndIceScreen", function(len)
+    CloseIceOverlay()
+end)
+
+net.Receive("BeginDimScreen", function(len)
+    local amount = net.ReadUInt(6)
+
+    if amount == 100 then
+        StartBlindOverlay()
+    else
+        StartDarkOverlay(amount)
+    end
+end)
+
+net.Receive("EndDimScreen", function(len)
+    EndDarkOverlay()
+end)
+
+hook.Add("TTTTutorialRoleText", "SummonerTutorialRoleText", function(playerRole)
+    local function getStyleString(role)
+        local roleColor = ROLE_COLORS[role]
+        return "<span style='color: rgb(" .. roleColor.r .. ", " .. roleColor.g .. ", " .. roleColor.b .. ")'>"
+    end
+
+    if playerRole == ROLE_ELEMENTALIST then
+        local divStart = "<div style='margin-top: 10px;'>"
+        local styleEnd = "</span>"
+
+        local html = "The " .. ROLE_STRINGS[ROLE_ELEMENTALIST] .. " is a member of the " .. getStyleString(ROLE_TRAITOR) .. "traitor team" .. styleEnd .. " whose goal is to eliminate all innocents and independents."
+
+        html = html .. divStart .. "They have access to " .. getStyleString(ROLE_ELEMENTALIST) .. "special powerups" .. styleEnd .. " in the " .. getStyleString(ROLE_TRAITOR) .. "traitor shop" .. styleEnd .. " which do a variety of things when they shoot other terrorists.</div>"
+
+        html = html .. divStart .. "The powerups available are:<ul style='margin-bottom: 0px; padding-bottom: 0px;'>"
+
+        local ROLE = ROLE_DATA_EXTERNAL[ROLE_ELEMENTALIST]
+        local allowUpgradedEffects = ROLE.ConvarTierUpgrades:GetBool()
+
+        if ROLE.ConvarPyroUpgrades:GetBool() then
```

Comment:
It might be worth putting a short description of each effect here for clarify

Reply:
The description might start getting raaaather lengthy with descriptions for each. The upgrades in the shop each have a description for the effect, having them here might be somewhat redundant. IDK.